fdbbackup/FileDecoder.actor.cpp

```diff
@@ -215,7 +281,44 @@ std::string decode_hex_string(std::string line) {
 	return line.substr(0, i);
 }
 
-int parseDecodeCommandLine(DecodeParams* param, CSimpleOpt* args) {
+// Parses and returns a ";" separated HEX encoded strings. So the ";" in
+// the string should be escaped as "\;".
+// Sets "err" to true if there is any parsing error.
+std::vector<std::string> parsePrefixesLine(const std::string& line, bool& err) {
+	std::vector<std::string> results;
+	err = false;
+
+	int p = 0;
+	while (p < line.size()) {
+		int end = line.find_first_of(';', p);
+		if (end == line.npos) {
+			end = line.size();
+		}
+		auto prefix = decode_hex_string(line.substr(p, end - p), err);
+		if (err) {
+			return results;
+		}
+		results.push_back(prefix);
+		p = end + 1;
+	}
+	return results;
+}
+
+std::vector<std::string> parsePrefixFile(const std::string& filename, bool& err) {
+	std::ifstream t(filename);
+	std::string line;
+
+	t.seekg(0, std::ios::end);
+	line.reserve(t.tellg());
+	t.seekg(0, std::ios::beg);
+
+	line.assign((std::istreambuf_iterator<char>(t)), std::istreambuf_iterator<char>());
```

Comment:
Does this require that the prefix file only contains one line?

Reply:
yes. the \r\n and \n could easily make mistakes if we are not careful. So I avoided using \n as the separator.